flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/Schema.java

```diff
@@ -665,13 +687,31 @@ public int hashCode() {
             this.dataType = dataType;
```

Comment:
We can share single constructor implementation by calling `this(columnName, dataType, null)`.